### PR TITLE
SIGINT-2040:  Dividing the common interface according to Job specific functionalities

### PR DESCRIPTION
**Pull request by maksudur-rahman-maruf**
- Modified the SecurityScan Interface.
- Added three new interfaces PrcommentScan, FreestyleScan, ReturnStatusScan. PrcommentScan contains only prcomment-specific methods, FreestyleScan contains only freestyle-specific methods and ReturnStatusScan contains only return-status specific method.
- Used PrcommentScan, FreestyleScan and ReturnStatusScan in the ScanParametersFactory class.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/extension/PrcommentScan.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package io.jenkins.plugins.synopsys.security.scan.extension;
+
+public interface PrcommentScan {
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/extension/PrcommentScan.java:3]: interface name can be modified to `PrCommentScan`

**Review by zaman-akib — CHANGES_REQUESTED** [review]
@maksudur-rahman-maruf  `isReturn_status()` in `SecurityScan` interface is not supported in FreeStyle job. Please do the refactoring for this method also.

**Review by zaman-akib — APPROVED** [review]

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/factory/ScanParametersFactory.java [hunk 2]
```diff
@@ -436,38 +441,38 @@ private static void prepareBlackDuckToolConfigurationParametersMap(
...
             polarisParametersMap.put(
-                    ApplicationConstants.COVERITY_CONFIG_PATH_KEY, securityScan.getPolaris_sast_config_path());
+                    ApplicationConstants.COVERITY_CONFIG_PATH_KEY, freestyleScan.getPolaris_sast_config_path());
         }
 
-        if (!Utility.isStringNullOrBlank(securityScan.getPolaris_sast_args())) {
-            polarisParametersMap.put(ApplicationConstants.COVERITY_ARGS_KEY, securityScan.getPolaris_sast_args());
+        if (!Utility.isStringNullOrBlank(freestyleScan.getPolaris_sast_args())) {
+            polarisParametersMap.put(ApplicationConstants.COVERITY_ARGS_KEY, freestyleScan.getPolaris_sast_args());
         }
```
Comment (jahid1209) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/factory/ScanParametersFactory.java:476]: This method is not covered in unit tests. Is it possible to add some?

Comment (maksudur-rahman-maruf) [reply src/main/java/io/jenkins/plugins/synopsys/security/scan/factory/ScanParametersFactory.java:476]: Unit tests for this method are not covered during the arbitrary feature implementation PR. Maybe we can add them in our next feature implementation.

Comment (jahid1209) [reply src/main/java/io/jenkins/plugins/synopsys/security/scan/factory/ScanParametersFactory.java:476]: You can add the tests in PR #147

**Pull request merged**